src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,121 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final double INCREASING_SIZE = 1.5;
+    private static final int START_POSITION = 0;
+    private static final int STEP_SIZE = 1;
+    private static final String INDEX_OUT_OF_BOUNDS_MESSAGE =
+            "Index %d out of bounds for size: %d";
+    private int size;
+    private Object[] array = new Object[DEFAULT_CAPACITY];
+
     @Override
     public void add(T value) {
-
+        if (size < array.length) {
+            array[size] = value;
+            size++;
+        } else {
+            increaseArray();
+            add(value);
+        }
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index >= 0 && index <= size) {
+            if (size >= array.length) {
+                increaseArray();
+            }
+            shiftArrayWithAddingElement(value, index);
+        }
+        checkIndex(index);
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        return (T) array[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        checkIndex(index);
+        array[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+        size--;
+        T removedElement = (T) array[index];
+        shiftArray(index);
+        return removedElement;
     }
 
     @Override
     public T remove(T element) {
-        return null;
+        int indexOfRemovedElement = 0;
+        T removedElement;
+        for (Object o : array) {
+            if (element == o || (element != null && element.equals(o))) {
+                removedElement = remove(indexOfRemovedElement);
+                return removedElement;
+            }
+            indexOfRemovedElement++;
+        }
+        throw new NoSuchElementException("No such element " + element + " in the list");
```

Comment:
Please be consistent, if you want to make exception messages constants, make them constants everywhere